extensions-core/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/taskadapter/PodTemplateTaskAdapter.java

```diff
@@ -264,6 +264,7 @@ private Map<String, String> getJobLabels(KubernetesTaskRunnerConfig config, Task
     return ImmutableMap.<String, String>builder()
         .putAll(config.getLabels())
         .put(DruidK8sConstants.LABEL_KEY, "true")
+        .put(DruidK8sConstants.ORIGINAL_NAMESPACE_KEY, config.getOverlordNamespace())
```

Comment:
does it work well when the namespace is undefined?

Reply:
Under KubernetesTaskRunnerConfig, the namespace is annotated as non-null. Hence, we should be able to get the namespace.